File: coauthor-for-vs-code/src/agent/ToolConfig.ts
/** Configuration interface for controlling tool behavior and automation features. */
export interface ToolConfig {
  usePrefillFromInput: boolean;
  autoExtractFigure: boolean;
  autoExtractTikzFigure: boolean;
  autoExtractTikzFigureReflect: boolean;
  attachTeXCount: boolean;
  autoConfirmation: boolean;
  printInputPrompt: boolean;
}

File: coauthor-python/coauthor/agent/tool_config.py
"""Tool configuration data structures."""

from dataclasses import dataclass


@dataclass
class ToolConfig:
    """Configuration for tool usage and automation features."""

    usePrefillFromInput: bool = False
    autoExtractFigure: bool = False
    autoExtractTikzFigure: bool = False
    autoExtractTikzFigureReflect: bool = False
    attachTeXCount: bool = False
    autoConfirmation: bool = False
    printInputPrompt: bool = False
    useOpenRouter: bool = False
